src/main/java/EdgeFileParser.java

```diff
@@ -0,0 +1,251 @@
+          return"";
+       }
+       currentLine = br.readLine().trim(); //this should be Text
+        String text = currentLine.substring(currentLine.indexOf("\"") + 1, currentLine.lastIndexOf("\"")).replaceAll(" ", ""); //get the Text parameter
+        if (text.equals("")) {
+          log.error("Blank names provided in diagram");
+          JOptionPane.showMessageDialog(null, "There are entities or attributes with blank names in this diagram.\nPlease provide names for them and try again.");
+          EdgeConvertGUI.setReadSuccess(false);
+          return"Error";
+       }
+       int escape = text.indexOf("\\");
+       if (escape > 0) { //Edge denotes a line break as "\line", disregard anything after a backslash
+          text = text.substring(0, escape);
+       }
+
+       do { //advance to end of record, look for whether the text is underlined
+          currentLine = br.readLine().trim();
+          if (currentLine.startsWith("TypeUnderl")) {
+             isUnderlined = true;
+          }
+       } while (!currentLine.equals("}")); // this is the end of a Figure entry
+
+       if (isEntity) { //create a new EdgeTable object and add it to the alTables ArrayList
+          if (isTableDup(text)) {
+             log.error("There are multiple tables with this name");
+             JOptionPane.showMessageDialog(null, "There are multiple tables called " + text + " in this diagram.\nPlease rename all but one of them and try again.");
+             EdgeConvertGUI.setReadSuccess(false);
+             return"Error";
+          }
+          alTables.add(new EdgeTable(numFigure + DELIM + text));
+       }
+       if (isAttribute) { //create a new EdgeField object and add it to the alFields ArrayList
+           EdgeField tempField = new EdgeField(numFigure + DELIM + text);
+          tempField.setIsPrimaryKey(isUnderlined);
+          alFields.add(tempField);
+       }
+       //reset flags
+       isEntity = false;
+       isAttribute = false;
+       isUnderlined = false;
+    }
+    return"";
+}
+    public void openFile(File inputFile){
+        int numLine = 0;
+        try {
+            log.info("Opening file: {}", inputFile.getName());
+            FileReader fr = new FileReader(inputFile);
+            br = new BufferedReader(fr);
+            //test for what kind of file we have
+            String currentLine = br.readLine().trim();
+            numLine++;
+            if (currentLine.startsWith(EDGE_ID)) { //the file chosen is an Edge Diagrammer file
+                parseEdgeFile(); //parse the file
+                br.close();
+                makeArrays(); //convert ArrayList objects into arrays of the appropriate Class type
+                resolveConnectors(); //Identify nature of Connector endpoints
+            }else { //the file chosen is something else
+                JOptionPane.showMessageDialog(null, "Unrecognized file format");
+                log.warn("Unrecognized file format");
+            }
```

Comment:
For easier readability, sometimes I like to use extract method for logic inside of try/catch blocks like this.